### ISSUE TITLE
no-use-before-define doesn't work

### ISSUE THREAD

**Issue by mefellows** [babel/babel-eslint#192]
The `no-use-before-define` rule doesn't work when the `babel-eslint` parser is used.

Steps to reproduce:

foo.js
```js
console.log(x); // Expect no-use-before-define to throw an error
var x = 1;
```

.eslintrc
```json
{
  "parser": "babel-eslint",
  "rules": {
    "no-use-before-define": 2
  }
}
```
Running 
```
> /tmp/foo $ eslint --config .eslintrc foo.js && echo $?
0
```

Note that removing `"parser": "babel-eslint"` results in the expected behaviour (error is thrown).

Versions:
* eslint version 1.6.0
* babel-eslint version 4.1.3

**Comment by hzoo** [comment]
Ok actually just talked about this in the eslint repo. Workaround for now is to use the `ecmaFeatures` config option https://github.com/eslint/eslint/blob/master/lib/rules/no-use-before-define.js#L98-L104
```js
  "ecmaFeatures": {
    "modules": true
  }
```

**Comment by mefellows** [comment]
Awesome thanks @hzoo, this workaround works nicely for now.

**Comment by hzoo** [comment]
I think this is the best thing to do so closing
